src/run_releasenotes_check.py

```diff
@@ -103,8 +105,156 @@ def create_urls_file_if_not_exists(manifest_files: List[InputManifest]) -> None:
+                            
+                            if all_tags_result.returncode == 0:
+                                tags = all_tags_result.stdout.strip().split('\n')
+                                logging.info(f"Available tags: {tags[:10]}")  # Show first 10 tags
+                                
+                                # Look for exact match or similar version
+                                for tag in tags:
+                                    if baseline_version in tag:
+                                        # Get date for this tag
+                                        tag_date_cmd = f"git log -1 --format=%ai {tag}"
+                                        tag_date_result = subprocess.run(tag_date_cmd, shell=True, capture_output=True, text=True, cwd=repo_dir)
+                                        
+                                        if tag_date_result.returncode == 0:
+                                            date_str = tag_date_result.stdout.strip().split()[0]
+                                            logging.info(f"Found tag date for {tag}: {date_str}")
+                                            return date_str
+                            
+                            raise Exception(f"Tag {baseline_version} not found in repository")
+                            
+                except Exception as e:
+                    logging.warning(f"Failed to get tag date via git: {e}")
+                
+                # If we get here, we couldn't find the tag date
+                raise ValueError(f"Could not find tag date for version {baseline_version}. Please use --date to specify the start date.")
+                
+            except ValueError:
+                # Re-raise ValueError as-is
+                raise
+            except Exception as e:
+                raise ValueError(f"Failed to determine baseline date: {e}. Please use --date to specify the start date.")
+        
+        # Get baseline date - use provided date or get from GitHub tag
+        current_version = manifests[0].build.version
+        if args.date:
+            baseline_date = str(args.date)
+            logging.info(f"Using provided baseline date: {baseline_date}")
+        else:
+            try:
+                baseline_date = get_baseline_date_from_github_tag(current_version)
+            except ValueError as e:
+                logging.error(str(e))
+                return 1
+        
+        logging.info(f"Generating AI release notes for version {current_version} since {baseline_date}")
+        
+        # Create ReleaseNotes instance with baseline date
+        release_notes = ReleaseNotes(manifests, baseline_date, args.action)
+        
+        # Generate AI release notes for each component
+        for i, manifest in enumerate(manifests):
+            manifest_path = args.manifest[i].name if i < len(args.manifest) else None
+            for component in manifest.components.select():
```

Comment:
This should be  `for component in manifest.components.select(focus=components, platform='linux'):` where.

Reply:
components is `args.components`.